docs/android/testing/integration_testing.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "Android integration testing"
+sidebar_label: "Integration testing"
+---
+
+🚧🚧🚧 Under Construction 🚧🚧🚧
+
+## Why perform integration testing?
+
+[Unit tests](unit_testing) are great and should be your primary choice when writing tests. However, integration testing ensures that the behavior of the application in a real Android environment is validated. Integration tests run on a real Android OS through an emulator, using the same JVM that will be used by end users.
+
+### Testing on a real JVM
+
+Some libraries we use have specific requirements, such as a minimum Java version above 8. Since our application targets Android API 21, which uses Java 8, these libraries cannot always be used. 
+
+During development, you might only test on the latest Android OS version, which uses a Java version above 8. If no one catches this during the review process, users on Android API 21 may encounter crashes.
+
+A concrete example is the [Jackson library](https://github.com/FasterXML/jackson). Starting from version 2.14, it requires a minimum Android version of 26. Unfortunately, such errors only arise at runtime, so the only way to catch them is through instrumentation tests. You can see the concrete example on this [PR](https://github.com/home-assistant/android/pull/5108).
```

Comment:
You were talking about Java versions and 8 / >8 but then you're talking about API levels here (not Android version)? The underlying cause of the issue is Java but that may not be obvious unless you look into that specific issue.

Reply:
I need to double check 👍🏻

Reply:
I rephrased the whole section.